projects/plugins/crm/src/automation/class-base-step.php

```diff
@@ -98,25 +108,62 @@ public function set_attribute_definitions( array $attribute_definitions ) {
 	}
 
 	/**
-	 * Set the next step.
+	 * Get the next step.
+	 *
+	 * Unless anything else is defined, then we assume to only continue with the
+	 * next step if the current one is successful.
+	 * One example of this will be conditions where a certain criteria has not been met.
 	 *
 	 * @since $$next-version$$
 	 *
-	 * @param int|string|null $step_id The next linked step id.
+	 * @return int|string|null The next linked step id.
 	 */
-	public function set_next_step( $step_id ) {
-		$this->next_step = $step_id;
+	public function get_next_step_id() {
+		return $this->get_next_step_true();
 	}
 
 	/**
-	 * Get the next step.
+	 * Get the next step if the current one is successful.
 	 *
 	 * @since $$next-version$$
 	 *
 	 * @return int|string|null The next linked step id.
 	 */
-	public function get_next_step_id() {
-		return $this->next_step_id;
+	public function get_next_step_true() {
+		return $this->next_step_true;
+	}
+
+	/**
+	 * Set the next step if the current one is successful.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return int|string|null The next linked step id.
+	 */
+	public function set_next_step_true() {
+		return $this->next_step_true;
+	}
+
+	/**
+	 * Get the next step if the current one is falsy.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return int|string|null The next linked step id.
+	 */
+	public function get_next_step_false() {
+		return $this->next_step_false;
+	}
+
+	/**
+	 * Set the next step if the current one is falsy.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return int|string|null The next linked step id.
+	 */
+	public function set_next_step_false() {
```

Comment:
This doesn't appear to set anything?

Reply:
Where's my "whistling emoji" when I need one - nice catch! Thank you!

_Context: They're there for the argument of always having matching setters when we have getters, but currently not used anywhere, so that's a nice catch._